src/components/Button/Button.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ButtonItem from './styles';
+
+const Button = ({ text, type, variant}) => (
+  <ButtonItem className={variant} type={type} onClick={() => console.log('A method')}>
```

Comment:
You don't need `className` because of styled-components. For `onClick` you can pass `onClick` prop.